src/auth/authenticator/tls/authenticator.go

```diff
@@ -0,0 +1,61 @@
+package tls
+
+import (
+	"bytes"
+	"crypto/x509"
+	"github.com/consensys/quorum-key-manager/pkg/errors"
+	"github.com/consensys/quorum-key-manager/src/auth/authenticator"
+	"github.com/consensys/quorum-key-manager/src/auth/types"
+	"net/http"
+)
+
+const AuthMode = "Tls"
+
+type Authenticator struct {
+	Certificates []*x509.Certificate
+}
+
+var _ authenticator.Authenticator = Authenticator{}
+
+func NewAuthenticator(cfg *Config) (*Authenticator, error) {
+	if len(cfg.Certificates) == 0 {
+		return nil, nil
+	}
+
+	auth := &Authenticator{Certificates: cfg.Certificates}
+
+	return auth, nil
+}
+
+// Authenticate checks certs and retrieve user Info
+// CN field -> Username
+// Organisation -> Groups
+func (authenticator Authenticator) Authenticate(req *http.Request) (*types.UserInfo, error) {
+	// extract Certificate info from request if any
+	if len(req.TLS.PeerCertificates) > 0 {
```

Comment:
Do you mind to rewrite the oppositive condition here so it looks like:
```
if len(req.TLS.PeerCertificates) == 0 {
return nil, errors.UnauthorizedError("no cert found in request")
}
```

Actually our linting so be capable to catch this which optmize the improve the code readability
```

Reply:
Ok